### DISCUSSION TITLE
Why use WSL when developing smart contracts?

### DISCUSSION THREAD

**Discussion by JamJamV** [smartcontractkit/full-blockchain-solidity-course-py#1465] [Q&A] [answered]
Hi all,
Recently saw that it recommends using the linux env through WSL to create smart contracts. Why is this the case? In the tutorial for setting up a windows environment is says it will "Save a lot of headaches" but I'd like to know as a primarily windows developer what the pros are.

Thank you :)

**Comment by cromewar** [comment, marked as answer]
Hello @JamJamV 

Well windows have some problems with:

1. Managing libraries trough node.
2. managing virtual envs with python.

Using wsl is just a way to have an isolated box on where you are going to install your development stuff and does not affect your windows installation, as is easier to solve problems on WSL rather to deal with the same ones on windows.

At the end is just a question of preference, if you want to develop using just windows is your choice but for example.

Web3 library needs something called Cytools so the difference between WSL and Windows are:

1. In WSL you don't have to do anything else that install the library.
2. In windows you will have to install Visual Studio (community) and also install the whole Python devellopment kit and also some .Net libraries.

Is jut easier to use WSL and that's it.
